src/EditorFeatures/Test/Diagnostics/DiagnosticsSquiggleTaggerProviderTests.cs

```diff
@@ -129,6 +134,19 @@ public void Dispose()
 
             await _asyncListener.CreateWaitTask();
         }
+
+        private class MyDiagnosticAnalyzerService : DiagnosticAnalyzerService
+        {
```

Comment:
the test used to use TestDiagnosticAnalyzerService which was created for certain unit tests that tests DiagnosticAnalyzerService itself. since it was testing itself, it didn't care about listener since it was purely unit tests. now this test tried to re-use this mock in integration style unit tests which now require listener, rather than re-using one created for another purpose, I created one specifically for this unit test.